src/deadline/blender_adaptor/BlenderClient/render_handlers/default_blender_handler.py

```diff
@@ -218,3 +219,35 @@ def set_scene_file(self, data: dict):
         if not os.path.isfile(file_path):
             raise FileNotFoundError(f"The scene file '{file_path}' does not exist")
         bpy.ops.wm.open_mainfile(filepath=file_path)
+
+    def set_render_device(self, data: dict) -> None:
+        """
+        Enables GPU rendering on the worker if specified in the user's configuration.
+
+        Args:
+            data (dict): The data given from the Adaptor. Keys expected: ['render_device']
+
+        Raises:
+            RuntimeError: If no devices of the specified type are available
+        """
+
+        if not data["render_device"] == "NONE":
```

Comment:
What do you think about doing an early return here? I generally find less indents easier to read and conditionals without negatives easier to reason about. IMO it also makes it easier to see that when the device is `NONE` nothing is supposed to happen. This is more of a stylistic thing, so it's up to you.

```python
if data["render_device"] == "NONE":
    return

cycles_preferences = bpy.context.preferences.addons["cycles"].preferences
cycles_preferences.refresh_devices()
devices = cycles_preferences.devices

if not devices:
    raise RuntimeError(
        f"Device type {data['render_device']} is not available on this worker."
    )

for device in devices:
    if device.type == "CPU":
        device.use = False
    else:
        device.use = True
        print(f"Enabled GPU device: {device.name}")

cycles_preferences.compute_device_type = data["render_device"]
print(f"Set render device to {data['render_device']}")
bpy.context.scene.cycles.device = "GPU"
```